packages/block-editor/src/components/block-list/block.js

```diff
@@ -341,6 +341,33 @@ function BlockListBlock( {
 					wrapper.current.focus();
 				}
 				break;
+			case TAB:
+				// In Edit mode, Tab should focus the first tabbable element
+				// after the content, which is the sidebar (with block controls)
+				// and Shift+Tab should focus the first tabbable element before
+				// the content, which is the block toolbar.
+				// Arrow keys can be used, and Tab and arrow keys can be used in
+				// Navigation mode (press Esc), to navigate through blocks.
+				if ( ( isSelected || isFirstMultiSelected ) && isEditMode ) {
+					if ( shiftKey ) {
+						if ( target === wrapper.current ) {
+							const editorElement = wrapper.current.parentElement.closest( '[tabindex]' );
+							const beforeEditorElement = focus.tabbable.findPrevious( editorElement );
+							beforeEditorElement.focus();
+							event.preventDefault();
+						}
+					} else {
+						const tabbables = focus.tabbable.find( wrapper.current );
+
+						if ( target === last( tabbables ) ) {
+							const editorElement = wrapper.current.parentElement.closest( '[tabindex]' );
```

Comment:
So if I understand properly, this is saying skip all the blocks?

Reply:
This probably breaks when you have parent blocks.

Reply:
You're right. We need another way to get the root element.

Reply:
Ideally all this tabbing behavior is handled in the same place (maybe FocusCapture)

Reply:
Moved it to WritingFlow, which fixes the issue.